engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/FlinkStreamingProcessRegistrar.scala

```diff
@@ -261,18 +267,27 @@ object FlinkStreamingProcessRegistrar {
 
     val enableObjectReuse = config.getOrElse[Boolean]("enableObjectReuse", true)
     val eventTimeMetricDuration = config.getOrElse[FiniteDuration]("eventTimeMetricSlideDuration", 10.seconds)
-    val checkpointInterval = config.as[FiniteDuration]("checkpointInterval")
+
+    // TODO checkpointInterval is deprecated - remove it in future
+    val checkpointInterval = config.getAs[FiniteDuration](path = "checkpointInterval")
+    if(checkpointInterval.isDefined) {
+      logger.warn("checkpointInterval config property is deprecated, use checkpointConfig.checkpointInterval instead")
+    }
+
+    val checkpointConfig = config.getAs[CheckpointConfig](path = "checkpointConfig")
+      .orElse(checkpointInterval.map(CheckpointConfig(_)))
+      .getOrElse(throw new IllegalArgumentException(s"At least checkpointInterval should be defined in process configuration"))
```

Comment:
WDYT about make this also optional? Doesn't Flink has some defaults for that? If it doesn't, we can introduce our default in code (10 seconds will be suitable) and not enforce our users to add this in their configs.

Reply:
I made it optional - without value checkpoint are not enabled for process (unless checkpoint interval is set in process properties)